docs/project/development-guidelines.md

```diff
@@ -0,0 +1,247 @@
+        const ret = await this.knex('users').select('*')
+         .where('nickname', nickname)
+
+        if (!ret[0] || ret[0].length == 0) return undefined
+        return User.fromJSON(ret[0])
+    }
+}
+
+```
+
+## Handling usecase failures
+
+To carry out the treatment of errors that occurred during the execution of the usecase, enable the environment variable HERBS_EXCEPTION in the production environments so that Herbs can obfuscate the errors generated and handle it on its return.
+
+``` jsx
+// Don't
+const createProduct = injection =>
+    usecase('Create Product', {
+
+        request: {
+            name: String,
+            ...
+        },
+
+        response: {
+            product: Product
+        },
+
+        'Check if the Product is valid': step(ctx => {
+           try {
+                if (!isValid) return Err(errors)
+                return Ok()
+            }
+            catch(e) {
+                throw new "Error to validated product"
+            }
+        }),
+
+        'Save the Product to the repository': step(async ctx => {
+            try {
+                ctx.ret.product = await repo.insert(product)
+                return Ok()
+            }
+            catch(e) {
+                throw new "Error to save product"
+            }
+        }),
+```
+
+``` jsx
+// Do
+
+HERBS_EXCEPTION = audit
+
+const createProduct = injection =>
+    usecase('Create Product', {
+
+        request: {
+            name: String,
+            ...
+        },
+
+        response: {
+            product: Product
+        },
+
+        'Check if the Product is valid': step(ctx => {
+            if (!isValid) return Err(errors)
+            return Ok()
+        }),
+
+        'Save the Product to the repository': step(async ctx => {
+            ctx.ret.product = await repo.insert(product)
+            return Ok()
+        }),
+```
+
+
+## Creating the business rules workflow
+
+Use `IfElse` to direct your flow of business rules, making your code easier to understand and document.
+
+``` jsx
+// Don't
+
+const updateItem = (injection) =>
+  usecase('Update Task', {
+
+    ...
+    
+    'Check if it is necessary to update Task positions': step(ctx => {
+        if(ctx.hasChangedPosition)
+        ...
+        else
+        ...
+    }),
+
+    }),
+```
+
+``` jsx
+// Do
+
+const updateItem = (injection) =>
+  usecase('Update Task', {
+
+    ...
+    
+    'Check if is necessary to update Task positions': ifElse({
+
+      'If position has changed': step(ctx => {
+        return Ok(ctx.hasChangedPosition) // true or false
+      }),
+
+      'Then rearrange positions and save Tasks on repository': step(async ctx => {
+        ...
+        return Ok()
+      }),
+
+      'Else save updated Task on repository': step(async ctx => {
+        ...
+        return Ok()
+      }),
+
+    }),
+```
+
+## Creating custom errors
+
+Use Herbs errors to generate metadata for the controller and thus customize your API return.
```

Comment:
Suggestion:
User Known Errors to expose runtime metadata. With this runtime information the glues like Herbs2REST or Herbs2GraphQL are able to change the API return.